overlays/pin-stackage.nix

```diff
@@ -1,5 +1,19 @@
-_: super: {
+self: super:
+
+let
+  sources = import ../nix/sources.nix;
+  stackage = sources.stackage;
+  stackageSourceJSON = self.writeText "stackage-source.json" (
```

Comment:
Wasn't too proud of this, but we can put it in a lib and be not proud of it once if we have to do it again.